src/recommender.py

```diff
@@ -36,36 +39,137 @@ def _process_evt(self, speaker_id, evt):
 
-    reward, _ = self.mock_scenario.insight(0, ctx, action_idx)
-
-    return reward
-
-
-  def _send_action(self, action):
+    time_count = 0
+    reward = ""
+
+
+    #determine variablename =
+    if ((action_idx + 1) >= 0) and ((action_idx +1) <=9):
+      var_name_code = "00"+str(action_idx+1)
+    elif ((action_idx + 1) >= 10) and ((action_idx +1) <=99):
+      var_name_code = "0"+str(action_idx+1)
+    elif ((action_idx + 1) >= 100) and ((action_idx +1) <=999):
+      var_name_code = str(action_idx+1)
+
+    # recieving reward from user
+    while time.time() - current_time < 30:
+      # print(str(empathid))
+      query = "SELECT answer FROM ema_data where primkey = '1:" + empathid + "' AND variablename = 'R" + var_name_code + "Q01'"
+      data = cursor.execute(query)
+
+      # if the user took some action
+      if data:
+        # empathid = '999|' + str(int(time.time()))
+        answer = str(cursor.fetchall()).split("'")[1]
+        # print('answer:', answer)
+
+        # if NO return 1
+        if answer == '2':
+          reward = 0.0
+          time_count += 1
+          break
+        # if YES return 0
+        if answer == '1':
+          reward = 1.0
+          time_count += 1
+          break
+
+    if time_count == 0:
+      err = "Timeout Error"
+    print(reward)
+
+    # close database
+    db.close()
+    return err, reward
+
+  def _send_action(self, speaker_id, action):
     '''
     Send the chosen action to the downstream
     return err if any
     '''
 
     err = None
-
-    # TODO actutal logic to send the data
-    print('Action', action)
-
-
-    # return err if any
-    return err
-
-
+    answer = 0;
+
+    # connect to database
+    db = pymysql.connect('localhost', 'root', '', 'ema')
+    cursor = db.cursor()
+
+    # start time
+    current_time = time.time()
+    last_time = 0
+    struc_current_time = time.localtime(current_time)
+
+    # the time must be between 8:00am and 12:00pm
+    if current_time - last_time >= 300 and (struc_current_time[3] < 25 and struc_current_time[3] > 8):
+      last_time = current_time
+
+
+      action = str(action)
+
+      survey_id = {'0':'19', '1':'20', '2':'21'}
+      #this should be 19 through 21
+
+      #survey_id = str(action + 19)  # each action plus 19
+      # add a dictionary
+
+
+      # items needed in url
+      pre_empathid = '999|' + str(int(time.time()))
+      empathid = None
+
+      phone_url = 'http://191.168.0.106:2226'
+      server_url = 'http://191.168.0.109/ema/ema.php'
+      androidid = 'db7d3cdb88e1a62a'
+      alarm = 'true'
+
+      # sending action to phone
+      try:
+        #send prequestion 22
+        url = phone_url + '/?q={%22id%22:%22' + str(speaker_id) + '%22,%22c%22:%22startsurvey%22,%22suid%22:%22' + '22' + '%22,%22server%22:%22' + server_url + '%22,%22androidid%22:%22' + androidid + '%22,%22empathid%22:%22' + pre_empathid + '%22,%22alarm%22:%22' + alarm + '%22}'
+        webbrowser.open(url)  # to open on browser
```

Comment:
so we still open a browser to request? do you want to change it to an `http GET` request?